HW2/P3/mandelbrot.pyx

```diff
@@ -0,0 +1,197 @@
+                #c = in_coords[i, j]
+                #
+                # Initialize other things
+                z_imag = AVX.float_to_float8(0.)
+                z_real = AVX.float_to_float8(0.)
+                four = AVX.float_to_float8(4.)
+                zero = AVX.float_to_float8(0.)
+
+                # An unchanging AVX with all ones, used several places
+                ones = AVX.float_to_float8(1.)
+
+                # Stores the iteration count for each of the 8 values
+                iter_count = AVX.float_to_float8(0.)
+
+                # 8 bits to store which data is still under 4 magnitude
+                under_four = AVX.float_to_float8(-1)
+
+                for iter in range(max_iterations):
+                    #with gil:
+                      # print_AVX(under_four)
+                      # print "Now for the signs"
+                      # print AVX.signs(under_four)
+                      # raw_input()
+                    # If all our values are greater than four then break
+                    if AVX.signs(under_four) == 0:
+                      break
+
+                    # Increment the iteration count only for those that are under four
+                    iter_count = AVX.add(iter_count, AVX.bitwise_and(under_four, ones))
+
+                    # Store values for z^2
+                    temp_real = AVX.sub(
+                      AVX.mul(z_real, z_real),
+                      AVX.mul(z_imag, z_imag)
+                    )
+                    temp_imag = AVX.add(
+                      AVX.mul(z_real, z_imag),
+                      AVX.mul(z_real, z_imag)
+                    )
+
+                    # Update z with z^2 + c
+                    z_real = AVX.add(temp_real, c_real)
+                    z_imag = AVX.add(temp_imag, c_imag)
+
+                    # with gil:
+                    #   print_AVX(z_real)
+                    #   print_AVX(z_imag)
+
+                    # Calculate the magnitude
+                    mag = AVX.add(
+                      AVX.mul(z_real, z_real),
+                      AVX.mul(z_imag, z_imag)
+                    )
+                    # with gil:
+                    #   print "Here"
+                    #   print_AVX(over_four)
+                    under_four = AVX.less_than(mag, four)
+                    
+                    # with gil:
+                    #   print_AVX(under_four)
+                    #   raw_input()
+                    # with gil:
+                    #   print "Magnitude!"
+                    #   print_AVX(mag)
+                    #   print "Over four"
+                    #   print_AVX(over_four)
+                    #   print AVX.signs(over_four)
+                    #   raw_input()
+                    #if magnitude_squared(z) > 4:
+                    #    break
+                    #z = z * z + c
+                #out_counts[i, j] = iter
+                counts_to_output(iter_count, out_counts, i, j)
+
```

Comment:
I executed my code in a very similar fashion. Good job keeping everything readable.